#! /bin/sh /usr/share/dpatch/dpatch-run
## opt-59_vdr-remote_instant_recordings.dpatch by 
## Frank Schmirler <vdrdev@schmirler.de>
## vdr-plugin-remotetimers
##
## All lines beginning with `## DP:' are a description of the patch.

@DPATCH@
diff -urNad vdr-1.7.15-orig/config.h vdr-1.7.15-a/config.h
--- vdr-1.7.15-orig/config.h	2010-07-01 17:45:16.223652000 +0200
+++ vdr-1.7.15-a/config.h	2010-07-01 17:56:57.559556299 +0200
@@ -42,6 +42,8 @@
 
 #define MAINMENUHOOKSVERSNUM 1.0
 
+#define REMOTEINSTANTVERSION 1.0
+
 #define MAXPRIORITY 99
 #define MAXLIFETIME 99
 
diff -urNad vdr-1.7.15-orig/menu.c vdr-1.7.15-a/menu.c
--- vdr-1.7.15-orig/menu.c	2010-07-01 17:45:16.231629000 +0200
+++ vdr-1.7.15-a/menu.c	2010-07-01 17:59:57.819137269 +0200
@@ -33,6 +33,7 @@
 #include "timers.h"
 #include "transfer.h"
 #include "videodir.h"
+#include "remotetimers.h"
 #include "menuorgpatch.h"
 #include "childlock.h"
 
@@ -4775,6 +4776,50 @@
 
 bool cRecordControls::Start(cTimer *Timer, bool Pause)
 {
+  if (!Timer) {
+     cTimer *t = new cTimer(true, Pause);
+
+     //get event
+     cSchedulesLock SchedulesLock;
+     const cEvent *event = NULL;;
+     const cSchedules *Schedules = cSchedules::Schedules(SchedulesLock);
+     if (Schedules) {
+        const cSchedule *Schedule = Schedules->GetSchedule(t->Channel());
+        if (Schedule) {
+           event = Schedule->GetEventAround(t->StartTime() + INSTANT_REC_EPG_LOOKAHEAD);
+        }
+     }
+
+     if (cPluginManager::GetPlugin("timersync")) {
+        // sets the file name
+        cRecording recording(t, event);
+        Timers.Add(t);
+        Timers.SetModified();
+        if (!cReplayControl::LastReplayed())
+           cReplayControl::SetRecording(recording.FileName(), recording.Name());
+        return true;
+     }
+     else {
+        RemoteTimers_InstantRecording_v1_0 ir;
+        ir.timer = t;
+        ir.pause = Pause;
+        ir.event = event;
+        ir.name = NULL;
+        ir.fileName = NULL;
+ 
+        if (cPluginManager::CallFirstService("RemoteTimers::InstantRecording-v1.0", &ir)) {
+           if (!cReplayControl::LastReplayed())
+              cReplayControl::SetRecording(ir.fileName, ir.name);
+ 	  delete t;
+ 	  // empty names indicate an error
+ 	  return *ir.fileName && *ir.name;
+        }
+        // service disabled or an error occured and the user confirmed local recording
+        // fall through to local instant recording
+     }
+     delete t;
+  }
+
   static time_t LastNoDiskSpaceMessage = 0;
   int FreeMB = 0;
   if (Timer) {
diff -urNad vdr-1.7.15-orig/remotetimers.h vdr-1.7.15-a/remotetimers.h
--- vdr-1.7.15-orig/remotetimers.h	1970-01-01 01:00:00.000000000 +0100
+++ vdr-1.7.15-a/remotetimers.h	2010-07-01 18:01:01.380634660 +0200
@@ -0,0 +1,25 @@
+/*
+ * remotetimers.h: Public interface of the plugin's services
+ *
+ * See the README file for copyright information and how to reach the author.
+ */
+
+#ifndef _SERVICE__H
+#define _SERVICE__H
+
+#ifndef __TIMERS_H
+#include <vdr/timer.h>
+#include <vdr/epg.h>
+#endif
+
+struct RemoteTimers_InstantRecording_v1_0 {
+//in
+	const cTimer	*timer;
+	bool		pause;
+	const cEvent	*event;
+//out
+	cString		name;
+	cString		fileName;
+};
+
+#endif //_SERVICE__H
